### ISSUE TITLE
TypeInfo.h redesign

### ISSUE THREAD

**Issue by end2endzone** [end2endzone/libVariant#7]
The TypeInfo.h include file contains the TypeInfo class which is a template class. 

The code of the TypeInfo.h file should be redesigned more in the following way:
```cpp
template <typename T> const char * get_typename(const T & value) { return "unknown"; }
#define DECLARE_SPECIALIZATION(T) template<>const char * get_typename<T>(const T & value ) { return #T; }
DECLARE_SPECIALIZATION(              bool);
DECLARE_SPECIALIZATION(              char);
DECLARE_SPECIALIZATION(       signed char);
DECLARE_SPECIALIZATION(     unsigned char);
DECLARE_SPECIALIZATION(             short);
DECLARE_SPECIALIZATION(    unsigned short);
DECLARE_SPECIALIZATION(               int);
DECLARE_SPECIALIZATION(      unsigned int);
DECLARE_SPECIALIZATION(              long);
DECLARE_SPECIALIZATION(     unsigned long);
DECLARE_SPECIALIZATION(         long long);
DECLARE_SPECIALIZATION(unsigned long long);
DECLARE_SPECIALIZATION(             float);
DECLARE_SPECIALIZATION(            double);
DECLARE_SPECIALIZATION(       long double);
DECLARE_SPECIALIZATION(           wchar_t);
#undef DECLARE_SPECIALIZATION
```

**Comment by end2endzone** [comment]
Also changed the definition of the following types:
* `float`, `double` and `long double` should be signed.
* `bool` and `wchat_t` should be unsigned.

**Comment by end2endzone** [comment]
Fixed in d22d453330f68058acbbd5bb8c30035f7c387ebd
